src/custom/hooks/useMetadataDoc.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react'
+import { useActiveWeb3React } from 'hooks/web3'
+import { generateReferralMetadataDoc, AppDataDoc } from 'utils/metadata'
+
+export function useMetadataDoc() {
+  const { account } = useActiveWeb3React()
+  const [document, setDocument] = useState<AppDataDoc>(generateReferralMetadataDoc(account || ''))
```

Comment:
To support passing an empty string we should update the `generateReferralMetadataDoc` function to return the default `appDataDoc` object when the `referralAddress` param is empty.